js/util/util.js

```diff
@@ -454,3 +455,40 @@ exports.warnOnce = function(message) {
         warnOnceHistory[message] = true;
     }
 };
+
+/**
+ * Calculates centroid of all the points in the ring https://en.wikipedia.org/wiki/Centroid#Centroid_of_polygon
+ * and http://paulbourke.net/geometry/polygonmesh/javascript.txt
+ * @param {Array} ring of Point
+ * @returns {Array} [x,y] Centroid of all the points
+ * @private
+ */
+exports.centroid = function(ring) {
+    assert(ring !== null);
+    assert(ring.length > 0);
+
+    if (ring.length === 1) {
+        return [ring[0].x, ring[0].y];
+    }
+
+    var len = ring.length;
+
+    assert(ring[0].x === ring[len - 1].x);
+    assert(ring[0].y === ring[len - 1].y);
+
+    var cX = 0;
+    var cY = 0;
+    var area = 0;
+
+    for (var i = 0, j = len - 1, p1, p2, mul; i < len; j = i++) {
+        p1 = ring[i];
+        p2 = ring[j];
+        mul = (p2.x * p1.y - p1.x * p2.y);
+        area += mul / 2;
+        cX += (p2.x + p1.x) * mul;
+        cY += (p2.y + p1.y) * mul;
+    }
+    var f = 6 * area;
+
+    return [cX / f, cY / f];
+};
```

Comment:
If we use pole of inaccessibility algorithm, this one is no longer used, so should be removed, right?